chain/client/src/chunk_validation.rs

```diff
@@ -442,12 +442,12 @@ impl Client {
         let shard_uid = self.epoch_manager.shard_id_to_uid(shard_id, &epoch_id)?;
         let prev_chunk_height_included = prev_chunk_header.height_included();
 
-        // TODO(#9292): previous chunk is genesis chunk - consider proper
-        // result for this corner case.
-        // let prev_chunk_prev_hash = *prev_chunk_header.prev_block_hash();
-        // if prev_chunk_prev_hash == CryptoHash::default() {
-        //     return Ok(vec![]);
-        // }
+        let prev_chunk_prev_hash = *prev_chunk_header.prev_block_hash();
+        if prev_chunk_prev_hash == CryptoHash::default() {
+            return Err(Error::Other(format!(
```

Comment:
Just confirming, we are fine recording this as an error when we are not expecting any state witness to be produced?

Reply:
The error is just logged and dropped. But this is not great design anyway. I just don't want to change return type to Option only for that corner case. Will leave comment

Reply:
actually I missed that I drop this case earlier. so this code is useless